pkg/skaffold/docker/image.go

```diff
@@ -218,6 +218,24 @@ func (l *localDaemon) Push(ctx context.Context, out io.Writer, ref string) (stri
 		return "", errors.Wrapf(err, "getting auth config for %s", ref)
 	}
 
+	// Check if the local image has a digest from a previous push
+	// If the remote image has the same digest, there's no need
+	// to push.
+	image, _, err := l.apiClient.ImageInspectWithRaw(ctx, ref)
+	if err == nil {
+		if len(image.RepoDigests) == 1 {
```

Comment:
i remember we had a brief conversation about this somewhere else....should this be `if len(image.RepoDigests) > 0`? i have no idea when or why there could be more than one digest but maybe we should account for that anyway?

Reply:
Yeah. My point was that I have no idea either. I decided to play it self.  I’ll just go through the normal path if there are many.